addons/mail/static/src/discuss/call/common/call.js

```diff
@@ -75,15 +115,56 @@ export class Call extends Component {
         const invitationCards = [];
         const filterVideos = this.props.thread.showOnlyVideo && this.props.thread.videoCount > 0;
         for (const session of Object.values(this.props.thread.rtcSessions)) {
-            if (!filterVideos || session.videoStream) {
-                const data = {
+            if (!filterVideos || Object.values(session.videoStreams).length) {
+                let data = {
                     key: "session_" + session.id,
                     session,
                 };
                 if (session.raisingHand) {
                     raisingHandCards.push(data);
                 } else {
-                    sessionCards.push(data);
+                    const videoCount = Object.values(session.videoStreams);
+                    if (videoCount.length) {
+                        switch (videoCount.length) {
+                            case 1:
+                                data.videoStream = videoCount[0];
+                                sessionCards.push(data);
+                                if (session.id === this.rtc.state.selfSession.id) {
+                                    this.state.draggableCard = false;
+                                }
+                                break;
+                            case 2:
+                                if (session.id === this.rtc.state.selfSession.id) {
+                                    data.videoStream = session.videoStreams["screen"];
```

Comment:
Aren't the videos of the peers indexed by `RTCRtpTransceiver.mid`? I think that it's a bit confusing that streams can either be indexed by label and by mid. Ideally the indices are always the same "type".